build.gradle.kts

```diff
@@ -84,3 +85,21 @@ publishing {
         }
     }
 }
+
+configure<com.diffplug.gradle.spotless.SpotlessExtension> {
+    kotlin {
+        ktlint().editorConfigOverride(
+            mapOf(
+                "indent_size" to "4",
+                "disabled_rules" to "package-name"
+            )
+        )
+    }
+    kotlinGradle {
+        target("*.gradle.kts")
+        ktlint().editorConfigOverride(
+            mapOf("indent_size" to "4")
+        )
+    }
+    isEnforceCheck = false
```

Comment:
I would be good to enable spotless:
- apply spotless for every commit (git hook) for every user by default
- check formatting on CI
Would you mind to add such features?

Reply:
AFAIK there is not a way to force hooks upon clone so I prepared `src/etc/githook/commit-msg` instead
Added step `Check with Gradle` to `ci.yml`

Reply:
In the js/npm world there are https://www.npmjs.com/package/simple-git-hooks and https://www.npmjs.com/package/husky, which configure git hooks. 

I found https://github.com/jakemarsden/git-hooks-gradle-plugin. I think we can use it. I tested it and it works properly. Projects it not very popular, but has small codebase and has no dependencies.

Reply:
Btw, you added `spotlessCheck` to git hook. In `apply spotless for every commit (git hook) for every user by default` I thought about `spotlessApply`. But we do not have to add this in this PR. Using `spotlessApply` in git hook is more complicated. because it should add modified (fixed) filed to git index. Moreover it would be good apply spotless only to changed files.

Reply:
Added [plugin](https://github.com/jakemarsden/git-hooks-gradle-plugin) and configured pre-commit hook